nvtabular/workflow/executor.py

```diff
@@ -0,0 +1,247 @@
+            # Compute the node's output
+            if node.op:
+                try:
+                    # use input_columns to ensure correct grouping (subgroups)
+                    selection = node.input_columns.resolve(node.input_schema)
+                    output_df = node.op.transform(selection, input_df)
+
+                    # Update or validate output_df dtypes
+                    for col_name, output_col_schema in node.output_schema.column_schemas.items():
+                        col_series = output_df[col_name]
+                        col_dtype = col_series.dtype
+                        is_list = is_list_dtype(col_series)
+
+                        if is_list:
+                            col_dtype = list_val_dtype(col_series)
+
+                        output_df_schema = output_col_schema.with_dtype(
+                            col_dtype, is_list=is_list, is_ragged=is_list
+                        )
+
+                        if capture_dtypes:
+                            node.output_schema.column_schemas[col_name] = output_df_schema
+                        elif len(output_df):
+                            if output_col_schema.dtype != output_df_schema.dtype:
+                                raise TypeError(
+                                    f"Dtype discrepancy detected for column {col_name}: "
+                                    f"operator {node.op.label} reported dtype "
+                                    f"`{output_col_schema.dtype}` but returned dtype "
+                                    f"`{output_df_schema.dtype}`."
+                                )
+                except Exception:
+                    LOG.exception("Failed to transform operator %s", node.op)
+                    raise
+                if output_df is None:
+                    raise RuntimeError(f"Operator {node.op} didn't return a value during transform")
+            else:
+                output_df = input_df
+
+            # Combine output across node loop iterations
+
+            # dask needs output to be in the same order defined as meta, reorder partitions here
+            # this also selects columns (handling the case of removing columns from the output using
+            # "-" overload)
+            if output is None:
+                output = output_df[node_output_cols]
+            else:
+                output = concat_columns([output, output_df[node_output_cols]])
+
+        if additional_columns:
+            output = concat_columns([output, df[get_unique(additional_columns)]])
+
+        return output
+
+
+class MerlinDaskExecutor:
+    def __init__(self, client=None):
+        self._executor = MerlinPythonExecutor()
+
+        # Deprecate `client`
+        if client is not None:
+            set_client_deprecated(client, "Workflow")
+
+    def __getstate__(self):
+        # dask client objects aren't picklable - exclude from saved representation
+        return {k: v for k, v in self.__dict__.items() if k != "client"}
```

Comment:
❓ I'm wondering where the client attribute is being set on the object (that this code is trying exclude). I don't see a `self.client` in here. Could be something outside this module doing something I suppose. Not suggesting we remove this now since it was here before and to reduce risk it makes sense to keep.

Reply:
I think this is leftover from a version before I realized I should use `set_client_deprecated`, so it is likely safe to remove. I know from the process of writing this that NVT tests will fail when saving a Workflow if there's a non-serializable client attribute on this object, so if it's problematic to remove, we'll find out quickly.